### PR TITLE
CLN: inline indexing 1-liners

### PR DESCRIPTION
**Pull request by jbrockmendel**
The only 2-liners here are _convert_scalar_indexer and _convert_slice_indexer, which are effectively `axis = min(axis, self.ndim-1)`, but that is unnecessary since all places where these are called have axis < self.ndim.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -841,8 +841,9 @@ def _ixs(self, i: int, axis: int = 0):
         """
         return self._values[i]
 
-    def _slice(self, slobj: slice, axis: int = 0, kind=None) -> "Series":
-        slobj = self.index._convert_slice_indexer(slobj, kind=kind or "getitem")
+    def _slice(self, slobj: slice, axis: int = 0, kind: str = "getitem") -> "Series":
+        # kind is either "getitem" or "iloc"
```
Comment (jreback) [inline pandas/core/series.py]: assert?

**Review by jreback — COMMENTED** [review]
small comment

**Comment by jreback** [comment]
also needs rebase

**Comment by jbrockmendel** [comment]
rebased+green

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: updated with assertion

**Comment by jreback** [comment]
thanks

**Pull request merged**